src/In.ProjectEKA.HipService/DataFlow/DataFlow.cs

```diff
@@ -3,39 +3,59 @@ namespace In.ProjectEKA.HipService.DataFlow
     using System;
     using System.Threading.Tasks;
     using HipLibrary.Patient.Model;
+    using MessagingQueue;
 
     public class DataFlow : IDataFlow
     {
+        private readonly IMessagingQueueManager messagingQueueManager;
         private readonly IDataFlowRepository dataFlowRepository;
+        private readonly IConsentArtefactRepository consentArtefactRepository;
 
-        public DataFlow(IDataFlowRepository dataFlowRepository)
+        public DataFlow(IDataFlowRepository dataFlowRepository,
+            IConsentArtefactRepository consentArtefactRepository,
+            IMessagingQueueManager messagingQueueManager)
         {
             this.dataFlowRepository = dataFlowRepository;
+            this.consentArtefactRepository = consentArtefactRepository;
+            this.messagingQueueManager = messagingQueueManager;
         }
         
         public async Task<Tuple<HealthInformationResponse, ErrorRepresentation>> HealthInformationRequestFor(HealthInformationRequest request)
         {
-            if (!IsValidConsentArtefact(request.Consent))
+            var (artefact, error) = consentArtefactRepository.GetFor(request.Consent.Id);
+            if (error != null)
             {
-                var errorRepresentation = new ErrorRepresentation(new Error(ErrorCode.ContextArtefactIdNotFound,
-                    ErrorMessage.ContextArtefactIdNotFound));
-                return new Tuple<HealthInformationResponse, ErrorRepresentation>(null, errorRepresentation);
+                return new Tuple<HealthInformationResponse, ErrorRepresentation>(null,
+                    new ErrorRepresentation(new Error(ErrorCode.ContextArtefactIdNotFound,
+                        ErrorMessage.ContextArtefactIdNotFound)));
             }
-            
+            var dataFlowArtefact = new DataFlowArtefact(artefact.CareContexts, request.HiDataRange,
```

Comment:
Can we call this DataFlowRequest/ DataRequest?

Reply:
Also add transaction Id

Reply:
done